common/nexus/incoming_service_registry.go

```diff
@@ -36,21 +36,22 @@ import (
 	"go.temporal.io/server/common"
 	"go.temporal.io/server/common/backoff"
 	"go.temporal.io/server/common/dynamicconfig"
-	"go.temporal.io/server/common/future"
 	"go.temporal.io/server/common/headers"
 	"go.temporal.io/server/common/log"
 	"go.temporal.io/server/common/log/tag"
 	p "go.temporal.io/server/common/persistence"
 	"go.temporal.io/server/internal/goro"
 )
 
+var errNexusAPIsDisabled = serviceerror.NewFailedPrecondition("error Nexus APIs are disabled.")
```

Comment:
I used not found in the frontend APIs.

https://github.com/temporalio/temporal/blob/c32abc3e7d8ce4c4fefcfa420f836703d4dc4d2c/service/frontend/nexus_http_handler.go#L128
https://github.com/temporalio/temporal/blob/c32abc3e7d8ce4c4fefcfa420f836703d4dc4d2c/service/frontend/operator_handler.go#L841

We should stay consistent.

But even better, don't do these checks in `waitUntilInitialized`, put the gate at the top of the handler.

Reply:
I know it's not necessary because at the moment this will only be accessed by the operator handler which already has these checks, but I like the little bit of extra defensiveness since we already need the feature flag to control whether the registry is polling for updates.

Reply:
Actually, this check really isn't necessary and might be confusing. I'll remove it.